nilrt_snac/_logging.py

```diff
@@ -0,0 +1,366 @@
+        # This captures all print() and subprocess output
+        sys.stdout = _TeeStream(original_stdout, log_file)
+        sys.stderr = _TeeStream(original_stderr, log_file)
+
+        # Redirect all existing logging handlers to use the new stderr stream
+        # This ensures logger output is interleaved with print() output
+        for handler in logging.root.handlers[:]:  # [:] creates a copy to iterate safely
+            if isinstance(handler, logging.StreamHandler):
+                # Save the original stream so we can restore it later
+                original_handler_streams.append((handler, handler.stream))
+                # Redirect handler to the new tee'd stderr
+                handler.stream = sys.stderr
+
+        # Yield the log path to the caller
+        yield log_path
+
+    except FileExistsError:
+        # This should be rare, but handle the case where the file already exists
+        raise SNACError(
+            f"Log file {log_path} already exists. This may indicate a clock issue or rapid re-execution.",
+            Errors.EX_ERROR,
+        )
+    except PermissionError as e:
+        raise SNACError(
+            f"Permission denied creating log file {log_path}: {e}", Errors.EX_BAD_ENVIRONMENT
+        )
+    except OSError as e:
+        raise SNACError(f"Failed to create log file {log_path}: {e}", Errors.EX_BAD_ENVIRONMENT)
+    finally:
+        # Restore original streams for all logging handlers
+        for handler, original_stream in original_handler_streams:
+            try:
+                handler.stream = original_stream
+            except Exception:
+                pass  # Ignore errors during restoration
+
+        # Restore original stdout and stderr
+        sys.stdout = original_stdout
+        sys.stderr = original_stderr
+
+        # Close log file if it was opened
+        if log_file is not None:
+            try:
+                # Write footer if we got this far
+                _write_log_footer(log_file, 0)  # We don't know the real return code here
+                log_file.close()
+            except Exception:
+                pass  # Ignore errors during cleanup
+
+        # Restore original streams (even in case of errors)
+        sys.stdout = original_stdout
+        sys.stderr = original_stderr
+
+        # Close log file if it was opened
+        if log_file is not None:
+            try:
+                # Write footer if we got this far
+                _write_log_footer(log_file, 0)  # We don't know the real return code here
+                log_file.close()
+            except Exception:
+                pass  # Ignore errors during cleanup
+
+
+def print_log_location(log_path: Optional[Path]) -> None:
+    """Print the location of the log file to the user.
+
+    Args:
+        log_path: Path to the log file, or None if logging was disabled
+    """
+    if log_path is not None:
+        print(f"\nLog saved to: {log_path}")
```

Comment:
This function seems like it should just be a final step in the `logging_context()`.